src/erlcloud_kinesis.erl

```
@@ -406,6 +407,71 @@ put_record(StreamName, PartitionKey, Data, ExplicitHashKey, Ordering, Config) wh
+%% Kinesis API:
+%% [https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecords.html]
+%%
+%% ===Example===
+%%
+%% This operation puts data records into an Amazon Kinesis stream from a producer.
+%%
+%% `
+%% erlcloud_kinesis:put_records(<<"StreamName">>, [{<<"data">>, <<"partitionKey1">>}]).
+%% {ok, [{<<"FailedRecordCount">>, 0},
+%%      {<<"Records">>,
+%%       [
+%%        [{<<"SequenceNumber">>,
+%%          <<"49537292605574028653758531131893428543501381406818304001">>},
+%%         {<<"ShardId">>,<<"shardId-000000000000">>}]
+%%       ]}
+%%     ]
+%% }
+%% erlcloud_kinesis:put_records(<<"StreamName">>, [{<<"data">>, <<"explicitHashKey">>, <<"partitionKey1">>}]).
+%% {ok, [{<<"FailedRecordCount">>, 0},
+%%      {<<"Records">>,
+%%       [
+%%        [{<<"SequenceNumber">>,
+%%          <<"49537292605574028653758531131893428543501381406818304001">>},
+%%         {<<"ShardId">>,<<"shardId-000000000000">>}]
+%%       ]}
+%%     ]
+%% }
+%% '
+%%
+%% @end
+%%------------------------------------------------------------------------------
+
+-type put_records_item() :: {Data :: string(), PartitionKey :: string()}
+                          | {Data :: string(), ExplicitHashKey :: string(), PartitionKey :: string()}.
+-type put_records_items() :: [put_records_item()].
+
+-spec put_records(string(), put_records_items()) -> proplist().
+
+put_records(StreamName, Items) ->
+    put_records(StreamName, Items, default_config()).
+
+-spec put_records(string(), put_records_items(), Config) -> proplist() when
+      Config :: aws_config().
+
+put_records(StreamName, Items, Config) ->
+    Operation = put_records_operation(),
+    Json = prepare_put_records_data(StreamName, Items),
+    erlcloud_kinesis_impl:request(Config, Operation, Json).
+
+put_records_operation() ->
+    "Kinesis_20131202.PutRecords".
+
+prepare_put_records_data(StreamName, Items) ->
+    Records = [prepare_put_records_item(X) || X <- Items],
```

@arkdro @nalundgaard API allows to put only 500 records in the batch? 
should we protect naive user by any means or let it fail with extremely descriptive `400 - InvalidArgumentException` 
in general in the library, we tent to protect, including this module.